### PR TITLE
Minor adjustments to workflows, dependencies and .ignores

### PR DESCRIPTION
**Pull request by TendTo**
- some dependencies bumps and introduction of dev dependecies
- some tests improvements (not really used)
- updated .dockerignore and .gitignore
- added auto-documentation to the workflow

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -36,30 +36,22 @@ def event_loop():
...
-
-
-@pytest.fixture(scope="session")
-async def bot(bot_initialize):
-    """Called at the beginning of the testing session.
-    Starts the bot with the testing setting in another thread
+    for test_key in config_map['test']:
+        if test_key in config_map:
+            config_map[test_key] = config_map['test'][test_key]
+        if test_key in config_map['meme']:
```
Comment (Helias) [inline tests/conftest.py:51]: should be `elif`?
can happen that both if are triggered in the same execution?

File: tests/test_db.py [hunk 2]
```diff
@@ -3,110 +3,87 @@
...
+        """Tests the query_from_string function for the database
+        """
+        DbManager.query_from_string("DROP TABLE IF EXISTS temp;", "CREATE TABLE temp(id int NOT NULL, PRIMARY KEY (id));",
+                                    "DROP TABLE temp;")
 
-def test_get_db(db_results):
-    """Tests the get_db function for the database
-    """
-    conn, cur = DbManager.get_db()
+        assert True
```
Comment (Helias) [inline tests/test_db.py:23]: :thinking:

File: tests/test_db.py [hunk 3]
```diff
@@ -3,110 +3,87 @@
...
-
-    assert count == 3
-
-    DbManager.delete_from(table_name="temp")
-    count = DbManager.count_from(table_name="temp", where="id >= %s", where_args=(-1, ))
-
-    assert count == 0
-
-    DbManager.query_from_string("""DROP TABLE temp;""")
+        DbManager.query_from_string("""DROP TABLE temp;""")
```
Comment (Helias) [inline tests/test_db.py:89]: don't need triple quotes

File: tests/conftest.py (hunk 1 above)
Comment (TendTo) [reply tests/conftest.py:51]: Nope, they can and will probably happen both

**Review by Helias — APPROVED** [review]

**Pull request merged**